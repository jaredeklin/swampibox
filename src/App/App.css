h1 {
  text-align: center;
  font-size: 90px;
  background-position: 90px;
  color: #FFFF00;
}

header {
  background-image: url('./star-gun.gif'), url('./star-gun.gif');
  background-repeat: no-repeat;
  background-position: 5%, 95%;
}

.error {
  text-align: center;
  margin-top: 30%;
}
