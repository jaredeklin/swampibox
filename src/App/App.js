import React, { Component } from 'react';
import './App.css';
import { CardContainer } from '../CardContainer/CardContainer';
import { fetchFilmData, cleanFilmData, fetchPeopleData , fetchVehicleData, fetchPlanetData } from '../apiHelper';
import { Opening } from '../Opening/Opening';
import { ButtonContainer } from '../ButtonContainer/ButtonContainer'

class App extends Component {
  constructor() {
    super()
    this.state = {
      filmData: [],
      favorites: [],
      peopleData: [],
      vehicleData: [],
      planetData: [],
      active: []
    }
  }

  getData = (type) => {
    this.addToActive(type)
    if (type === 'People') {
      fetchPeopleData()
        .then(peopleData => this.setState({ peopleData }))
    }

    if (type === 'Vehicles') {
      fetchVehicleData()
        .then(vehicleData => this.setState({ vehicleData }))
    }

    if (type === 'Planets')
      fetchPlanetData()
        .then(planetData => this.setState({ planetData }))
  }

  addToActive = (type) => {
    const activeCard = [...this.state.active]
    if (activeCard.length >= 1 && activeCard[0] !== type) {
      activeCard.splice(0, 1, type)
    } else if (activeCard[0] !== type){
      activeCard.push(type)
    }
    this.setState({ active: activeCard})
  }


  addToFavorites = (event) => {
    // console.log(name, pop, species, homeworld)
  }


  componentDidMount() {
    // this.getData();
    // this.fetchPeopleData()
    
  }

  render() {
    return (
      <div className="App">
        <Opening filmData={this.state.filmData} />
        <header className="App-header">
          <h1 className="App-title">SWAMPIbox</h1>
          <button className="view-favorites-btn">favorites {this.state.favorites.length}</button>
        </header>
        
        <ButtonContainer getData={this.getData} active={this.state.active}/>
        
            <CardContainer 
              peopleData={this.state.peopleData}
              planetData={this.state.planetData}
              vehicleData={this.state.vehicleData} 
              addToFavorites={this.addToFavorites} 
              allData={this.state}
            />
        
       
        
      </div>
    );
  }
}

export default App;
