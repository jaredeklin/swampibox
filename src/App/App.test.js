import React from 'react';
// import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', ()=> {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with an empty state', () => {
    const expectedState = {
      filmData: [],
      favorites: [],
      peopleData: [],
      vehicleData: [],
      planetData: [],
      active: []
    }

    expect(wrapper.state()).toEqual(expectedState)
  });
});