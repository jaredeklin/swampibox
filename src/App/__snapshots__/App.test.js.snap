// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          filmData={Array []}
/>,
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    SWAMPIbox
          </h1>
          <button
                    className="view-favorites-btn"
          >
                    favorites 
                    0
          </button>
</header>,
        <Unknown
          active={Array []}
          getData={[Function]}
/>,
        false,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "filmData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              SWAMPIbox
</h1>,
            <button
              className="view-favorites-btn"
>
              favorites 
              0
</button>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWAMPIbox",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "SWAMPIbox",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "favorites ",
                0,
              ],
              "className": "view-favorites-btn",
            },
            "ref": null,
            "rendered": Array [
              "favorites ",
              0,
            ],
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": Array [],
          "getData": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            filmData={Array []}
/>,
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        SWAMPIbox
            </h1>
            <button
                        className="view-favorites-btn"
            >
                        favorites 
                        0
            </button>
</header>,
          <Unknown
            active={Array []}
            getData={[Function]}
/>,
          false,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filmData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                SWAMPIbox
</h1>,
              <button
                className="view-favorites-btn"
>
                favorites 
                0
</button>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWAMPIbox",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "SWAMPIbox",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "favorites ",
                  0,
                ],
                "className": "view-favorites-btn",
              },
              "ref": null,
              "rendered": Array [
                "favorites ",
                0,
              ],
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": Array [],
            "getData": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
